src/components/SideMenu/index.tsx

```diff
@@ -0,0 +1,44 @@
+import { useState } from 'react'
+import { Link } from 'react-router-dom'
+import classnames from 'classnames'
+
+import styles from './SideMenu.module.scss'
+
+function SideMenu() {
+  const [isOpen, setIsOpen] = useState(false)
+
+  const toggleMenu = () => {
+    setIsOpen((current) => !current)
```

Comment:
setIsOpen(!current) should work here, its not in any useEffect / useCallback etc